### PR TITLE
Use ruff to replace flake8 and plugins

### PR DESCRIPTION
**Pull request by justinchuby**
### Description

Use ruff https://github.com/charliermarsh/ruff to lint files, replacing flake8 and its plugins. It is faster and has auto-fix.

- Also enforce PEP8-naming
- Simplified the lint CI

### Motivation and Context

Code quality

### CODE REVIEW

**Review by jcwchen — APPROVED** [review]
Please fix DCO. To confirm, so current codebase does follow pep8-naming?

**Comment by justinchuby** [comment]
> Please fix DCO. To confirm, so current codebase does follow pep8-naming?

I think so! Although I am going to replace this with ruff

**Comment by justinchuby** [comment]
@jcwchen it good now

File: pyproject.toml [hunk 1]
```diff
@@ -70,3 +70,44 @@ disable = [
     "no-name-in-module",
     "useless-return",
 ]
+
+[tool.ruff]
+target-version = "py37"
```
Comment (jcwchen) [inline pyproject.toml:75]: Why only set the lowest supported Python version here?

File: onnx/tools/net_drawer.py [hunk 2]
```diff
@@ -61,18 +61,20 @@ def ReallyGetOpNode(op: NodeProto, op_id: int) -> pydot.Node:
             node.set_URL(url)
         return node
 
-    return ReallyGetOpNode
+    return really_get_op_node
 
 
-def GetPydotGraph(
+def get_pydot_graph(
```
Comment (jcwchen) [inline onnx/tools/net_drawer.py:67]: Not sure whether it is safe to change the public API name here. Possibly some users might rely on these APIs already?

File: pyproject.toml [hunk 3]
```diff
@@ -70,3 +70,44 @@ disable = [
...
+]
+ignore = [
+    "B9", # Opinionated bugbear rules
+    "E402", # warns on imports coming after statements. There are important use cases
+            # like demandimport (https://fburl.com/demandimport) that require statements
+    "E501",
+    "F405", # FIXME: is hard to silence since we indeed do star import
+    "N803", # Argument casing
+    "N812",
+    "N806",
```
Comment (jcwchen) [inline pyproject.toml:93]: Could you please also add comments for these 2 if possible? Thanks!

File: .github/workflows/lint.yaml [hunk 4]
```diff
@@ -6,34 +6,16 @@ on:
...
-          sudo apt-get install -qq -o=Dpkg::Use-Pty=0 -y --no-install-recommends dos2unix
-          git submodule update --init --recursive
-          export ONNX_BUILD_TESTS=0
-          export ONNX_ML=1
-          export CMAKE_ARGS="-DONNXIFI_DUMMY_BACKEND=ON -DONNX_WERROR=ON"
-          # ONNX lite
-          export CMAKE_ARGS="${CMAKE_ARGS} -DONNX_USE_LITE_PROTO=ON"
-          export ONNX_NAMESPACE=ONNX_NAMESPACE_FOO_BAR_FOR_CI
-
-          python setup.py install
```
Comment (jcwchen) [inline .github/workflows/lint.yaml:36]: To confirm, we don't need to build ONNX from source in the Lint CI anymore?

File: pyproject.toml (hunk 1 above)
Comment (justinchuby) [reply pyproject.toml:75]: This is so ruff doesn’t auto fix with styles Python 3.7 doesn’t support.

File: .github/workflows/lint.yaml (hunk 4 above)
Comment (justinchuby) [reply .github/workflows/lint.yaml:36]: Yes for the optional lints. Lintrunner still needs built onnx

File: onnx/tools/net_drawer.py (hunk 2 above)
Comment (justinchuby) [reply onnx/tools/net_drawer.py:67]: If you like I can revert this change ?

File: .github/workflows/lint.yaml (hunk 4 above)
Comment (justinchuby) [reply .github/workflows/lint.yaml:36]: https://github.com/charliermarsh/ruff#target-version

File: onnx/tools/net_drawer.py (hunk 2 above)
Comment (jcwchen) [reply onnx/tools/net_drawer.py:67]: Yes, I think it's safer. And please add a TODO comment to change it in next major version bump. Thank you!

File: pyproject.toml (hunk 3 above)
Comment (justinchuby) [reply pyproject.toml]: Updated and refined

File: onnx/tools/net_drawer.py (hunk 2 above)
Comment (justinchuby) [reply onnx/tools/net_drawer.py:67]: Done

Comment (justinchuby) [reply onnx/tools/net_drawer.py:67]: Actually, this is a commandline tool. I don't think anyone will use the functions themselves? If they do personally I think it's still a minor break?

Comment (justinchuby) [reply onnx/tools/net_drawer.py:67]: (also, tools/executables (anything with a `main()`) shouldn't be imported so implementation details are expected to change imo)

**Comment by justinchuby** [comment]
Looks like coverage changed after the model updates in ci: https://dev.azure.com/onnx-pipelines/onnx/_build/results?buildId=37519&view=logs&j=2105a913-e5a3-5f74-1c30-4f8b7a56f2db&t=c2afa4d8-3a4c-5e2d-1542-4328675067bd&l=85

**Comment by jcwchen** [comment]
> Looks like coverage changed after the model updates in ci: https://dev.azure.com/onnx-pipelines/onnx/_build/results?buildId=37519&view=logs&j=2105a913-e5a3-5f74-1c30-4f8b7a56f2db&t=c2afa4d8-3a4c-5e2d-1542-4328675067bd&l=85

It failed, because some of real models was not downloaded from download.onnxruntime.ai. The new URLs seem not reliable enough... I have rerun the pipelines and now they are all green.

File: onnx/tools/net_drawer.py (hunk 2 above)
Comment (justinchuby) [reply onnx/tools/net_drawer.py:67]: I see ok! Thanks for doing the search. I will revert in this case. I don't think it's worth changing this just for the naming. Although we should not have exposed functions in a file with `__main__`

**Review by jcwchen — APPROVED** [review]
LGTM. Thank you for introducing ruff!

**Comment by justinchuby** [comment]
> @justinchuby a question: are the changes from an automated tool, or do they include some manual changes? Asking since manual changes probably benefit from a careful review, unlike automated changes.

only naming changes are manual (pep8 naming)

**Review by gramalingam — APPROVED** [review]
LGTM, thanks!

**Comment by justinchuby** [comment]
@jcwchen something's wrong with the queue?

**Comment by justinchuby** [comment]
Looks like the workflows need more triggers

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge_group

https://github.com/community/community/discussions/39933

**Comment by justinchuby** [comment]
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-github-actions

**Comment by jcwchen** [comment]
Here is the status:
image

It's a little tricky -- for required pipeline running by GitHub action, adding `merge_group` should help. For other pipeline providers, for instance Azurepipelines, using some branch detection should help. However, for required DCO, I have no idea what we can do to make it pass... I will try to ask it in https://github.com/orgs/community/discussions/46757. It's still in beta so it might take some time to change and improve. For now, I am inclined to disable it, if there is no way to make DCO run in the queue.

**Pull request merged**